test/SimpleNameRegisterTest.t.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "ds-test/test.sol";
+import 'src/SimpleNameRegister.sol';
+
+interface CheatCodes {
+    function prank(address) external;
+    function expectRevert(bytes calldata) external;
+    function startPrank(address) external;
+    function stopPrank() external;
+}
+
+abstract contract StateZero is DSTest {
+    SimpleNameRegister public simpleNameRegister;
+    CheatCodes cheats;
+    
+    function setUp() public virtual {
+        simpleNameRegister = new SimpleNameRegister();
+        cheats = CheatCodes(HEVM_ADDRESS);
+    }
+}
+
+contract StateZeroTest is StateZero {
+
+    function testCannotRelease(string memory testString) public {  
+        cheats.expectRevert(bytes("Not your name!"));
+        simpleNameRegister.release(testString);
+    }
+
+    function testRegister(string memory testString) public {
+        simpleNameRegister.register(testString);
+        bool success = (address(this) == simpleNameRegister.holder(testString));
+        assertTrue(success);
+    }
+}
+
+abstract contract StateRegistered is StateZero {
+    address adversary;
+    string name;
+
+    function setUp() public override {
+        super.setUp();
+        adversary = 0xE6A2e85916802210147e366D4431f5ca4dD51a78;
+        
+        // state transition
+        name = 'whale';
+        simpleNameRegister.register(name);
+    }
+}
+
+contract StateRegisteredTest is StateRegistered {
+
+    function testAdversaryCannotRegisterName() public {
+        cheats.startPrank(adversary);
```

Comment:
use `prank()` for calls on one line, startPrank/stopPrank for multi line calls

Reply:
Ah, I had thought because `prank()` only modified the next function call, and the next line was `expectRevert`, it would have failed and not applied the prank effect on `simpleNameRegister.release(name),` which came after `expectRevert`.

Just tested and realized that this is not the case. Thanks for pointing it out my misconception👍